source/tests/system/nidmm_driver_api_tests.cpp

```diff
@@ -0,0 +1,279 @@
+            dmm::GetAttributeViInt32Response response;
+            ::grpc::Status status = GetStub()->GetAttributeViInt32(&context, request, &response);
+            EXPECT_TRUE(status.ok());
+            expect_api_success(response.status());
+            return response.attribute_value();
+        }
+
+        ViReal64 get_real64_attribute(const char* channel_name, dmm::NiDmmAttributes attribute_id)
+        {
+            ::grpc::ClientContext context;
+            dmm::GetAttributeViReal64Request request;
+            request.mutable_vi()->set_id(GetSessionId());
+            request.set_channel_name(channel_name);
+            request.set_attribute_id(attribute_id);
+            dmm::GetAttributeViReal64Response response;
+            ::grpc::Status status = GetStub()->GetAttributeViReal64(&context, request, &response);
+            EXPECT_TRUE(status.ok());
+            expect_api_success(response.status());
+            return response.attribute_value();
+        }
+
+    private:
+        std::shared_ptr<::grpc::Channel> channel_;
+        std::unique_ptr<::nidevice_grpc::Session> driver_session_;
+        std::unique_ptr<dmm::NiDmm::Stub> nidmm_stub_;
+        std::unique_ptr<nidevice_grpc::SessionRepository> session_repository_;
+        std::unique_ptr<dmm::NiDmmLibrary> nidmm_library_;
+        std::unique_ptr<dmm::NiDmmService> nidmm_service_;
+        std::unique_ptr<::grpc::Server> server_;
+};
+
+TEST_F(NiDmmDriverApiTest, SelfTest_CompletesSuccessfully)
+{
+  ::grpc::ClientContext context;
+  dmm::SelfTestRequest request;
+  request.mutable_vi()->set_id(GetSessionId());
+  dmm::SelfTestResponse response;
+
+  ::grpc::Status status = GetStub()->SelfTest(&context, request, &response);
+
+  EXPECT_TRUE(status.ok());
+  expect_api_success(response.status());
+  EXPECT_EQ(0, response.self_test_result());
+  EXPECT_LT(0, strlen(response.self_test_message().c_str()));
+}
+
+TEST_F(NiDmmDriverApiTest, Reset_CompletesSuccessfully)
+{
+  ::grpc::ClientContext context;
+  dmm::ResetRequest request;
+  request.mutable_vi()->set_id(GetSessionId());
+  dmm::ResetResponse response;
+
+  ::grpc::Status status = GetStub()->Reset(&context, request, &response);
+
+  EXPECT_TRUE(status.ok());
+  expect_api_success(response.status());
+}
+
+TEST_F(NiDmmDriverApiTest, SetViReal64Attribute_GetViReal64Attribute_ValueMatchesSetValue)
+{
+  const char* channel_name = "";
+  const dmm::NiDmmAttributes attribute_to_set = dmm::NiDmmAttributes::NIDMM_ATTRIBUTE_TRIGGER_DELAY;
+  const ViReal64 expected_value = 42.24;
+  ::grpc::ClientContext context;
+  dmm::SetAttributeViReal64Request request;
+  request.mutable_vi()->set_id(GetSessionId());
+  request.set_channel_name(channel_name);
+  request.set_attribute_id(attribute_to_set);
+  request.set_attribute_value(expected_value);
+  dmm::SetAttributeViReal64Response response;
+
+  ::grpc::Status status = GetStub()->SetAttributeViReal64(&context, request, &response);
```

Comment:
Move this up one line so it's part of the setup block. Same with the other tests.

Reply:
moved it